### PR TITLE
Implement Pre & Post Login and Registration Handlers

### LINKED ISSUE #576: Implement Pre & Post Login and Registration Handlers
As defined in the framework spec [here](https://github.com/stormpath/stormpath-framework-spec/blob/master/handlers.md).

### PR DESCRIPTION
**Pull request by jarias**
Fixes #576

### CODE REVIEW

File: examples/servlet/src/main/java/com/stormpath/sdk/examples/servlet/PostLoginHandler.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * @since 1.0.0
+ */
+public class PostLoginHandler implements WebHandler {
+    @Override
+    public boolean handle(HttpServletRequest request, HttpServletResponse response, Account account) {
+        System.out.println("----> PostLoginHandler");
```
Comment (mraible) [inline examples/servlet/src/main/java/com/stormpath/sdk/examples/servlet/PostLoginHandler.java]: Remove `System.out.println` or replace with `log.debug`.

File: examples/servlet/src/main/java/com/stormpath/sdk/examples/servlet/PostRegisterHandler.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * @since 1.0.0
+ */
+public class PostRegisterHandler implements WebHandler {
+    @Override
+    public boolean handle(HttpServletRequest request, HttpServletResponse response, Account account) {
+        System.out.println("----> PostRegisterHandler");
```
Comment (mraible) [inline examples/servlet/src/main/java/com/stormpath/sdk/examples/servlet/PostRegisterHandler.java]: Same as above, remove `System.out.println`. Applies to all `System.out.println`'s in this PR.

File: extensions/servlet/src/test/groovy/com/stormpath/sdk/servlet/mvc/LoginControllerTest.groovy [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        expect(request.getAttribute(UserAgents.USER_AGENT_REQUEST_ATTRIBUTE_NAME)).andReturn new DefaultUserAgent(request)
+        expect(request.getHeader("Accept")).andReturn "text/html"
+        expect(request.getParameter("next")).andReturn null
+
+        //Minimum things required to test the onValidSubmit method only
+        LoginController loginController = new LoginController(
+                preLoginHandler: loginPreHandler,
+                authenticationResultSaver: authenticationResultSaver,
+                produces: Arrays.asList(MediaType.TEXT_HTML)
+
```
Comment (mraible) [inline extensions/servlet/src/test/groovy/com/stormpath/sdk/servlet/mvc/LoginControllerTest.groovy]: Remove extra line break.

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathLoginSuccessHandler.java [hunk 4]
```diff
@@ -75,12 +81,19 @@ public void onAuthenticationSuccess(HttpServletRequest request, HttpServletRespo
...
         // Content Negotiation per https://github.com/stormpath/stormpath-sdk-java/issues/682
         try {
-            MediaType mediaType =
-                ContentNegotiationResolver.INSTANCE.getContentType(request, response, supportedMediaTypes);
-            if (MediaType.APPLICATION_JSON.equals(mediaType)) {
-                request.getRequestDispatcher(meUri).forward(request, response);
-            } else {
-                super.onAuthenticationSuccess(request, response, authentication);
+            boolean shouldContinue = true;
+
```
Comment (mraible) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathLoginSuccessHandler.java]: Remove extra line break.

**Comment by jarias** [comment]
@mraible PR updated

File: examples/servlet/src/main/webapp/WEB-INF/stormpath.properties [hunk 5]
```diff
@@ -182,3 +182,10 @@ stormpath.web.uris./dashboard = account
 stormpath.web.me.enabled = true
 stormpath.web.me.uri = /me
 stormpath.web.me.view = me
+
+
```
Comment (mraible) [inline examples/servlet/src/main/webapp/WEB-INF/stormpath.properties]: Remove extra line.

**Comment by mraible** [comment]
LGTM 👍

**Pull request merged**